app/Http/Controllers/SecondFactorController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Lara\Http\Controllers;
+
+
+use Illuminate\Http\RedirectResponse;
+use Lara\Http\Middleware\RejectGuests;
+
+use PragmaRX\Google2FA\Google2FA;
+use Redirect;
+
+class SecondFactorController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware(RejectGuests::class);
+    }
+
+    public function index()
+    {
+        return \View::make('secondfactor.index');
+    }
+
+    /** verifys current code
+     *
+     * @return RedirectResponse
+     */
+    public function verify()
+    {
+        $validator = \Validator::make(\Input::all(), [
+            'code' => 'required|digits:6'
+        ]);
+        if ($validator->fails()) {
+            return Redirect::back()->withErrors($validator->errors());
+        }
+        $targeturl = \Session::get('targeturl', "/");
```

Comment:
I'm not quite sure what this line does, what ware the contents of `$targeturl`?

Reply:
scenario:

1.  you are user
2. 2fa is enabled for your account
2.  you are not logged in
3.  you watch /event/1234
4.  you log in 
5. system redirects to 2fa
6. user sends the 2fa code
7. system redirects user back to /event/1234

Targeturl will stored in the middleware ;)